File: Sorting/linear.c
//linear search
#include<stdio.h>
#include<stdbool.h>
void linear(int arr[],int target,int n);
int main(){
    int n=5;
    int arr[]={2,3,4,1,8};
    linear(arr,4,n);
    return 0;

}
void linear(int arr[],int target,int n){
    bool found=false;
    for(int i=0;i<n;i++){
        if(arr[i]==target){
            printf("found %d at index : %d ",arr[i],i);
            found=true;
        }

    }
    if(!found){
        printf("no found !! sorry");
    }
   
}

File: 1-array/linear.java
import java.util.*;
public class linear {
    public static int search(int arr[],int target){
        int n=arr.length;
        for(int i=0;i<n;i++){
         
            if(arr[i]==target){
                System.out.println("found");
                return i;
            }
        
        }
       
        return -1;
    }
    public static void main(String args[]){
        int arr[]={1,4,2,7,5};
        System.out.print(search(arr,4));
    }
    
}
